### DISCUSSION TITLE
M48 probe test - strange results, always increasing

### DISCUSSION THREAD

**Discussion by dwery** [Jyers/Marlin#912] [Q&A] [answered]
Hello,
 I built a probe (tiny touch on thingiverse), and the M48 command is showing an always increasing Z value. At each iteration the detected Z will be a few tenths of mm higher than before. If I execute the command again, the first value would be higher than the last from the previous iteration. Mechanically the probe seems to be working nicely and the photointerrupter is being triggered when it should. Homing Z would restart the sequence from the lowest detected z.

Anyone observed something similar?

**Comment by Jyers** [comment, marked as answer]
Could perhaps be an issue with Z steps/mm or Z offset, that being said I don't have any experience with custom probes. All I can tell you is that M48 is most likely not the issue.

**Reply by dwery** [reply]
Thanks. I suspect an issue with the brass nut. Gonna change it and retry.

**Reply by dwery** [reply]
Changed the nut, rebuilt the axis. Problem seems gone, however I'm not 100% confident it was the nut and not some other issue with the gantry.

**Reply by ghost** [reply]
maybe check vref for Z, in case too low and losing steps?

**Reply by dwery** [reply]
@badbod Nope, looks like it has plenty of power and movement is smooth. I have to hold it real hard to have it loose a step.
